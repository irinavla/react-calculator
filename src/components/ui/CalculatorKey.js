
import React, { Component } from 'react'
import PointTarget from 'react-point'

class CalculatorKey extends Component {
  render() {
    const { onPress, className, ...props } = this.props
    
    return (
      <PointTarget onPoint={onPress}>
        <button className={`calculator-key ${className}`} {...props}/>
      </PointTarget>
    )
  }
}

export default CalculatorKey