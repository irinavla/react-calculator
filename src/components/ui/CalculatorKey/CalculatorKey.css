.operators {
  width: 25%;
  color: var(--white);
  background-color: var(--red-color);
  display: flex;
  flex-direction: column;
  font-size: 36px;
  font-weight: 400;
}

.operator { 
  height: 25%;
  flex:1;
  border-bottom: 1px solid var(--dark-grey-color);
}

.keys {
  width: 75%;
  background-color: var(--white);
  border-right: 1px solid var(--dark-grey-color);
  color: var(--dark-grey-color);
}

.numbers {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: 75%;

}

.key { 
  width: calc(33.33% - 1px);
  height: calc(33.33% - 1px);
  border-bottom: 1px solid var(--dark-grey-color);
  border-right: 1px solid var(--dark-grey-color);
  font-size: 45px;
  font-weight: 300;
}

.key:nth-child(3n) {
  border-right: none;
  width: 33.33%;
}

.clear { 
  height: calc(25% - 1px);
  background-color: var(--white);
  border-bottom: 1px solid var(--dark-grey-color);
  font-size: 23px;
  font-weight: 300;
}