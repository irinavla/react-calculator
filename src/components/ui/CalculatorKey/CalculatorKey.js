
import React, { Component } from 'react'
import PointTarget from 'react-point'
import './CalculatorKey.css';

class CalculatorKey extends Component {
  render() {
    const { onPress, className, ...props } = this.props
    
    return (
      <PointTarget onPoint={onPress}>
        <div className={`${className}`} {...props}></div>
      </PointTarget>
    )
  }
}

export default CalculatorKey