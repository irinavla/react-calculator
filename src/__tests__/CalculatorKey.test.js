import React from 'react';

import renderer from 'react-test-renderer'
import CalculatorKey from '../components/ui/CalculatorKey/CalculatorKey';


it('renders correctly', () => {
  const tree = renderer.create( <CalculatorKey />).toJSON();

  expect(tree).toMatchSnapshot();
});