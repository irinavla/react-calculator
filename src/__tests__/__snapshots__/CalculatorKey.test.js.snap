// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="undefined"
  onClick={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
/>
`;
